### PR TITLE
Pagination sorting swagger branch

### PR DESCRIPTION
**Pull request by qreqit**

### CODE REVIEW

File: src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java [hunk 1]
```diff
@@ -19,8 +20,8 @@ public  class BookController {
     private BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public Page<BookDto> getAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java:23]: Where Swagger annotations?

File: src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java [hunk 2]
```diff
@@ -19,8 +20,8 @@ public  class BookController {
...
+        return bookService.findAll(pageable);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java:24]: Add Pageable to searchBooks also

File: src/main/java/mate/academy/springbootwebgreqit/repository/BookPaginationAndSortingRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package mate.academy.springbootwebgreqit.repository;
+
+import mate.academy.springbootwebgreqit.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+public interface BookPaginationAndSortingRepository extends PagingAndSortingRepository<Book, Long> {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/repository/BookPaginationAndSortingRepository.java:9]: Where did you use this interface and methods?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java (hunk 1 above)
Comment (qreqit) [reply src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java:23]: i will fix it

File: src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java (hunk 2 above)
Comment (qreqit) [reply src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java:24]: ok

File: src/main/java/mate/academy/springbootwebgreqit/repository/BookPaginationAndSortingRepository.java (hunk 3 above)
Comment (qreqit) [reply src/main/java/mate/academy/springbootwebgreqit/repository/BookPaginationAndSortingRepository.java]: Oh, I thought that I needed this class, but now I know that `BookRepository` already has these methods

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**